Providers/Scripts/2.4x-2.5x/Scripts/nxOMSContainers.py

```diff
@@ -0,0 +1,284 @@
+        copy_all_files(reg_dir, REG_PATH)
+        check_and_create_yaml(yaml_dir, YAML_PATH)
+    elif Ensure == 'Absent':
+        # and delete CONF file in the directory
+        delete_all_files(conf_dir, CONF_PATH)
+    else:
+        # log error Ensure value not expected
+        LG().Log('ERROR', "Ensure value: " + Ensure + " not expected")
+        return [-1]
+    
+    # restart oms agent
+    if OMS_ACTION.restart_oms_agent():
+        return [0]
+    else:
+        return [-1]
+
+def Test(WorkspaceId, Ensure):
+    """
+    Test method for the DSC resoruce
+    If it returns [0] no further action is taken
+    If it returns [-1] Set_Marshall is called
+    """
+
+    # test for the existence of plugin and conf subfolders in the current plugin
+    plugin_dir = os.path.join(MODULE_RESOURCE_DIR, "plugin")
+    conf_dir = os.path.join(MODULE_RESOURCE_DIR, "conf")
+    lib_dir = os.path.join(MODULE_RESOURCE_DIR, "lib")
+    reg_dir = os.path.join(MODULE_RESOURCE_DIR, "reg")
+    yaml_dir = os.path.join(MODULE_RESOURCE_DIR, "yaml")
+    
+    if Ensure == 'Present':
+        # check all files exist
+        if (not check_all_files(plugin_dir, PLUGIN_PATH) or not check_all_files(conf_dir, CONF_PATH) or not check_all_files(lib_dir, LIB_PATH) or not check_all_files(reg_dir, REG_PATH) or not check_all_files(yaml_dir, YAML_PATH)):
+            return [-1]
+    elif Ensure == 'Absent':
+        return [-1];
+    else:
+        # log error Ensure value not expected
+        LG().Log('ERROR', "Ensure value: " + Ensure + " not expected")
+        return [-1]
+
+def copy_all_files(src, dest):
+    try:
+        src_files = os.listdir(src)
+        for file_name in src_files:
+            full_file_name = os.path.join(src, file_name)
+            if (os.path.isfile(full_file_name)):
+                shutil.copy(full_file_name, dest)
+    except:
+        LG().Log('ERROR', 'copy_all_files failed for src: ' + src + ' dest: ' + dest)
+        return False
+
+def check_and_create_yaml(src, dest):
+    try:
+        src_files = os.listdir(src)
+        for file_name in src_files:
+            full_file_name = os.path.join(src, file_name)
+            if (not os.path.isfile(full_file_name)):
+                shutil.copy(full_file_name, dest)
+                os.chmod(os.path.join(dest, file_name), 0644)
+                os.chown(os.path.join(dest, file_name), GetUID('omsagent'), GetGID('omsagent'))
+    except:
+        LG().Log('ERROR', 'copy_all_files failed for src: ' + src + ' dest: ' + dest)
+        return False
+            
+def delete_all_files(src, dest):
+    try:
+        src_files = os.listdir(src)
+        for file_name in src_files:
+            full_file_name = os.path.join(dest, file_name)
+            if (os.path.isfile(full_file_name)):
+                os.remove(full_file_name)
+    except:
+        LG().Log('ERROR', 'delete_all_files failed for src: ' + src + ' dest: ' + dest)
+        return False
+
+
+def check_all_files(src, dest):
```

Comment:
A lot of this code seems to be copied from nxOMSPlugin. Did you make sure it all applies for your usage, and you aren't creating dead code?

Reply:
There is no dead code. nxOMSContainers has some of the logic similar to nxOMSPlugin